Flaky beat.TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green

`beat.TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green` has failed on [EKS](https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-eks/134/testReport/github/com_elastic_cloud-on-k8s_test_e2e_beat/TestVersionUpgradeToLatest7x_ES_endpoint_should_eventually_be_reachable/), on [GKE](https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-stack-versions/191/testReport/github/com_elastic_cloud-on-k8s_test_e2e_beat/Run_tests_for_different_ELK_stack_versions_in_GKE___7_3_2___TestVersionUpgradeToLatest7x_ES_cluster_health_should_eventually_be_green/) and on [OCP](https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests-ocp/220/testReport/junit/github/com_elastic_cloud-on-k8s_test_e2e_beat/TestVersionUpgradeToLatest7x_ES_cluster_health_should_eventually_be_green/):

```
=== RUN   TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green
Retries (5m0s timeout): ....................................................................................................
    utils.go:84: 
        	Error Trace:	utils.go:84
        	Error:      	Received unexpected error:
        	            	health is yellow
        	Test:       	TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green
{"log.level":"error","@timestamp":"2020-09-01T01:38:04.537Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:43\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.RunMutationsWhileWatching\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/run_mutation.go:75\ngithub.com/elastic/cloud-on-k8s/test/e2e/beat.TestVersionUpgradeToLatest7x\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/beat/upgrade_test.go:53\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1108"}
    --- FAIL: TestVersionUpgradeToLatest7x/ES_cluster_health_should_eventually_be_green (300.00s)
```

```
=== RUN   TestVersionUpgradeToLatest7x/ES_endpoint_should_eventually_be_reachable
Retries (10m0s timeout): .......................................................................................................................................................................................................
    utils.go:84: 
        	Error Trace:	utils.go:84
        	Error:      	Received unexpected error:
        	            	cluster health is not green, but yellow
        	Test:       	TestVersionUpgradeToLatest7x/ES_endpoint_should_eventually_be_reachable
{"log.level":"error","@timestamp":"2020-09-01T02:12:24.737Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:43\ngithub.com/elastic/cloud-on-k8s/test/e2e/test.RunMutationsWhileWatching\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/run_mutation.go:75\ngithub.com/elastic/cloud-on-k8s/test/e2e/beat.TestVersionUpgradeToLatest7x\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/beat/upgrade_test.go:53\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1108"}
GET /_cat/shards?v
index                            shard prirep state       docs  store ip             node
ilm-history-2-000001             0     p      STARTED                 192.168.83.194 test-beat-upgrade-to-7x-6htj-es-masterdata-0
filebeat-7.8.1-2020.09.01-000001 0     p      STARTED    22117 10.3mb 192.168.83.194 test-beat-upgrade-to-7x-6htj-es-masterdata-0
filebeat-7.8.1-2020.09.01-000001 0     r      UNASSIGNED                             
 
GET /_cluster/allocation/explain
{"index":"filebeat-7.8.1-2020.09.01-000001","shard":0,"primary":false,"current_state":"unassigned","unassigned_info":{"reason":"INDEX_CREATED","at":"2020-09-01T02:02:23.568Z","last_allocation_status":"no_attempt"},"can_allocate":"no","allocate_explanation":"cannot allocate because allocation is not permitted to any of the nodes","node_allocation_decisions":[{"node_id":"9k97PQgtSm65O8DzRGFNvA","node_name":"test-beat-upgrade-to-7x-6htj-es-masterdata-0","transport_address":"192.168.83.194:9300","node_attributes":{"k8s_node_name":"ip-192-168-93-78.eu-central-1.compute.internal","ml.machine_memory":"2147483648","xpack.installed":"true","transform.node":"true","ml.max_open_jobs":"20"},"node_decision":"no","weight_ranking":1,"deciders":[{"decider":"same_shard","decision":"NO","explanation":"a copy of this shard is already allocated to this node [[filebeat-7.8.1-2020.09.01-000001][0], node[9k97PQgtSm65O8DzRGFNvA], [P], s[STARTED], a[id=MGnUX0mKTH6rGos7cL8R7Q]]"}]}]} 
    --- FAIL: TestVersionUpgradeToLatest7x/ES_endpoint_should_eventually_be_reachable (600.00s)
```

--- a/test/e2e/beat/upgrade_test.go
+++ b/test/e2e/beat/upgrade_test.go
@@ -29,7 +29,7 @@ func TestVersionUpgradeToLatest7x(t *testing.T) {
 
 	name := "test-beat-upgrade-to-7x"
 	esBuilder := elasticsearch.NewBuilder(name).
-		WithESMasterDataNodes(1, elasticsearch.DefaultResources).
+		WithESMasterDataNodes(3, elasticsearch.DefaultResources).
 		WithVersion(dstVersion)
 
 	fbBuilder := beat.NewBuilder(name).

